desc/magnetic_fields.py

```diff
@@ -782,16 +823,36 @@ def field_line_integrate(
     x0 = jnp.array([r0, phis[0] * jnp.ones_like(r0), z0]).T
 
     @jit
-    def odefun(rpz, s):
+    def odefun(s, rpz, args):
         rpz = rpz.reshape((3, -1)).T
         r = rpz[:, 0]
         br, bp, bz = field.compute_magnetic_field(rpz, params, basis="rpz").T
         return jnp.array(
             [r * br / bp * jnp.sign(bp), jnp.sign(bp), r * bz / bp * jnp.sign(bp)]
         ).squeeze()
 
-    intfun = lambda x: odeint(odefun, x, phis, rtol=rtol, atol=atol, mxstep=maxstep)
+    # diffrax parameters
+    stepsize_controller = PIDController(rtol=rtol, atol=atol)
+    terminating_event = (
+        DiscreteTerminatingEvent(terminating_event) if terminating_event else None
```

Comment:
It would be nice if we could make the default terminating event to stop if the magnetic field is undefined. Like if `field.compute_magnetic_field(rpz, params, basis="rpz")` returns NaNs because the integration has left the field's domain. 

That was my desired use case for this PR, but I'm not sure how to implement it.

Reply:
wasted the whole day trying to figure this out... hopefully the diffrax people can tell me https://github.com/patrick-kidger/diffrax/issues/290

Reply:
Sorry... but thanks for looking into it!